ReasoningEngine/GraphOperations/CommandProcessor.cs

```diff
@@ -15,7 +15,7 @@ public CommandProcessor(GraphFileManager graphFileManager)
             this.graphFileManager = graphFileManager;
         }
 
-        public string ProcessCommand(string command, string payload)
+        public virtual string ProcessCommand(string command, string payload)
```

Comment:
This isn't exactly a review but what difference does it make when using virtual on this? I'm not quite familiar with this

Reply:
It just allows for easier testing with the unit tests. The unit tests try to make a parameterless mock I think?
To be honest I'm not 100% clear why they are necessary, but when I didn't have the the unit tests would break.